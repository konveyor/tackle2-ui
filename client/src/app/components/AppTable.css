.pf-v5-c-table.app-table
  thead
  > tr
  > th.pf-v5-c-table__toggle
  > :first-of-type {
  display: none;
}
