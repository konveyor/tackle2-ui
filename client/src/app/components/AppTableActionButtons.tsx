import React from "react";
import { useTranslation } from "react-i18next";
import { Button, Flex, FlexItem } from "@patternfly/react-core";
import { applicationsWriteScopes, RBAC, RBAC_TYPE } from "@app/rbac";
import { ConditionalTooltip } from "./ConditionalTooltip";
import { Td } from "@patternfly/react-table";

export interface AppTableActionButtonsProps {
  isDeleteEnabled?: boolean;
  tooltipMessage?: string;
  onEdit: () => void;
  onDelete: () => void;
}

export const AppTableActionButtons: React.FC<AppTableActionButtonsProps> = ({
  isDeleteEnabled = false,
  tooltipMessage = "",
  onEdit,
  onDelete,
}) => {
  const { t } = useTranslation();

  return (
    <RBAC
      allowedPermissions={applicationsWriteScopes}
      rbacType={RBAC_TYPE.Scope}
    >
      <Td isActionCell>
        <Flex
          direction={{ default: "row" }}
          flexWrap={{ default: "nowrap" }}
          style={{ minWidth: "100%" }}
        >
          <FlexItem spacer={{ default: "spacerSm" }} />
          <FlexItem spacer={{ default: "spacerSm" }}>
            <Button
              id="edit-button"
              aria-label="edit"
              variant="secondary"
              onClick={onEdit}
            >
              {t("actions.edit")}
            </Button>
          </FlexItem>
          <FlexItem spacer={{ default: "spacerSm" }}>
            <ConditionalTooltip
              isTooltipEnabled={isDeleteEnabled}
              content={tooltipMessage}
            >
              <Button
                id="delete-button"
                aria-label="delete"
                variant="link"
                onClick={onDelete}
                isAriaDisabled={isDeleteEnabled}
              >
                {t("actions.delete")}
              </Button>
            </ConditionalTooltip>
          </FlexItem>
        </Flex>
      </Td>
    </RBAC>
  );
};
