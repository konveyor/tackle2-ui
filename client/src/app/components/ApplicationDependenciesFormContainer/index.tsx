import React from "react";
import { FormContextProvider } from "./FormContext";
import {
  ApplicationDependenciesFormProps,
  ApplicationDependenciesForm,
} from "./ApplicationDependenciesForm";

export const ApplicationDependenciesFormContainer: React.FC<
  ApplicationDependenciesFormProps
> = ({ ...props }) => {
  return (
    <FormContextProvider>
      <ApplicationDependenciesForm {...props} />
    </FormContextProvider>
  );
};
