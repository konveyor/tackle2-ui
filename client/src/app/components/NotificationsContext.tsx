import * as React from "react";
import { AlertProps } from "@patternfly/react-core";

export type INotification = {
  title: string;
  variant: AlertProps["variant"];
  message?: React.ReactNode;
  hideCloseButton?: boolean;
  timeout?: number | boolean;
};

interface INotificationsProvider {
  children: React.ReactNode;
}

interface INotificationsContext {
  pushNotification: (
    notification: INotification,
    clearNotificationDelay?: number
  ) => void;
  dismissNotification: (key: string) => void;
  notifications: INotification[];
}

const appContextDefaultValue = {} as INotificationsContext;

const notificationDefault: Pick<INotification, "hideCloseButton"> = {
  hideCloseButton: false,
};

export const NotificationsContext = React.createContext<INotificationsContext>(
  appContextDefaultValue
);

export const NotificationsProvider: React.FunctionComponent<
  INotificationsProvider
> = ({ children }: INotificationsProvider) => {
  const [notifications, setNotifications] = React.useState<INotification[]>([]);

  const pushNotification = (
    notification: INotification,
    clearNotificationDelay?: number
  ) => {
    setNotifications((prevNotifications) => [
      ...prevNotifications,
      { ...notificationDefault, ...notification },
    ]);

    setTimeout(() => {
      setNotifications((prevNotifications) => {
        return prevNotifications.filter(
          (notif) => notif.title !== notification.title
        );
      });
    }, clearNotificationDelay || 8000);
  };

  const dismissNotification = (title: string) => {
    const remainingNotifications = notifications.filter(
      (n) => n.title !== title
    );
    setNotifications(remainingNotifications);
  };

  return (
    <NotificationsContext.Provider
      value={{
        pushNotification,
        dismissNotification,
        notifications,
      }}
    >
      {children}
    </NotificationsContext.Provider>
  );
};
