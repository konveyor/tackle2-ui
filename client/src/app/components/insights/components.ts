export * from "./components/affected-apps-link";
export * from "./components/insight-description-and-links";
export * from "./components/insight-expanded-row-content";
