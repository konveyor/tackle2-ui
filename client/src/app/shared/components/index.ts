export { AppPlaceholder } from "./app-placeholder";
export { StateError } from "./app-table";
export { AppTableActionButtons } from "./app-table-action-buttons";
export { AppTableWithControls } from "./app-table-with-controls";
export { Color } from "./color";
export { ConditionalRender } from "./conditional-render";
export { ConfirmDialog } from "./confirm-dialog";
export { EmptyTextMessage } from "./empty-text-message";
export { KebabDropdown } from "./kebab-dropdown";
export { NoDataEmptyState } from "./no-data-empty-state";
export { PageHeader } from "./page-header";
export { ProposedActionLabel } from "./proposed-action-label";
export { RiskLabel } from "./risk-label";
export { SimpleEmptyState } from "./simple-empty-state";
export * from "./simple-select";
export * from "./iconed-status";
export { ToolbarBulkSelector } from "./toolbar-bulk-selector";
