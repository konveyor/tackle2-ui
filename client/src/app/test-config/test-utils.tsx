import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
