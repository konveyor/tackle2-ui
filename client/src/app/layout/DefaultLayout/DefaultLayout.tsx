import React from "react";
import { Page, SkipToContent } from "@patternfly/react-core";

import { HeaderApp } from "../HeaderApp";
import { SidebarApp } from "../SidebarApp";
import { Notifications } from "@app/components/Notifications";
import { PageContentWithDrawerProvider } from "@app/components/PageDrawerContext";

export interface DefaultLayoutProps {}

export const DefaultLayout: React.FC<DefaultLayoutProps> = ({ children }) => {
  const pageId = "main-content-page-layout-horizontal-nav";
  const PageSkipToContent = (
    <SkipToContent href={`#${pageId}`}>Skip to content</SkipToContent>
  );

  return (
    <Page
      header={<HeaderApp />}
      sidebar={<SidebarApp />}
      isManagedSidebar
      skipToContent={PageSkipToContent}
      mainContainerId={pageId}
    >
      <PageContentWithDrawerProvider>
        {children}
        <Notifications />
      </PageContentWithDrawerProvider>
    </Page>
  );
};
