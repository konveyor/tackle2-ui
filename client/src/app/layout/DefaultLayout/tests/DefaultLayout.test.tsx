import { render } from "@app/test-config/test-utils";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { DefaultLayout } from "../DefaultLayout";
import { NotificationsProvider } from "../../../components/NotificationsContext";

it.skip("Test snapshot", () => {
  const wrapper = render(
    <Router>
      <NotificationsProvider>
        <DefaultLayout />
      </NotificationsProvider>
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
