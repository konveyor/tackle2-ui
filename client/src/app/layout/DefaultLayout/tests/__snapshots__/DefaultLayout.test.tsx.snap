// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-c-page pf-m-resize-observer pf-m-breakpoint-default pf-m-height-breakpoint-sm"
      >
        <a
          class="pf-c-button pf-m-primary pf-c-skip-to-content"
          href="#main-content-page-layout-horizontal-nav"
        >
          Skip to content
        </a>
        <header
          class="pf-c-page__header"
        >
          <div
            class="pf-c-page__header-brand"
          >
            <div
              class="pf-c-page__header-brand-toggle"
            >
              <button
                aria-disabled="false"
                aria-expanded="false"
                aria-label="Global navigation"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                id="nav-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 448 512"
                  width="1em"
                >
                  <path
                    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  />
                </svg>
              </button>
            </div>
            <a
              class="pf-c-page__header-brand-link"
            >
              <img
                alt="brand"
                class="pf-c-brand"
                src="test-file-stub"
              />
            </a>
          </div>
          <div
            class="pf-c-page__header-tools"
          >
            <div
              class="pf-c-page__header-tools-group pf-m-hidden"
            >
              <div
                class="pf-c-page__header-tools-item"
              >
                <button
                  aria-disabled="false"
                  aria-label="about-button"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="aboutButton"
                  type="button"
                >
                  <test-file-stub />
                </button>
              </div>
            </div>
            <div
              class="pf-c-page__header-tools-group"
            >
              <div
                class="pf-c-page__header-tools-item"
              >
                <div
                  class="pf-c-dropdown pf-m-align-right"
                  data-ouia-component-id="OUIA-Generated-Dropdown-1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Actions"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    id="pf-dropdown-toggle-id-0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 192 512"
                      width="1em"
                    >
                      <path
                        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="pf-c-page__header-tools-item pf-m-hidden"
              />
            </div>
          </div>
        </header>
        <div
          aria-hidden="true"
          class="pf-c-page__sidebar pf-m-collapsed"
          id="page-sidebar"
        >
          <div
            class="pf-c-page__sidebar-body"
          >
            <nav
              aria-label="Nav"
              class="pf-c-nav"
              data-ouia-component-id="OUIA-Generated-Nav-1"
              data-ouia-component-type="PF4/Nav"
              data-ouia-safe="true"
              id="nav-primary"
            >
              <div
                class="perspective"
              >
                <div
                  class="pf-c-select"
                  data-ouia-component-id="OUIA-Generated-Select-single-1"
                  data-ouia-component-type="PF4/Select"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-label="Options menu"
                    aria-labelledby=" pf-select-toggle-id-1"
                    class="pf-c-select__toggle"
                    id="pf-select-toggle-id-1"
                    type="button"
                  >
                    <div
                      class="pf-c-select__toggle-wrapper"
                    >
                      <span
                        class="pf-c-select__toggle-text"
                      >
                        sidebar.developer
                      </span>
                    </div>
                    <span
                      class="pf-c-select__toggle-arrow"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <ul
                class="pf-c-nav__list"
                title="Global"
              >
                <li
                  class="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-1"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/applicationsundefined"
                    tabindex="-1"
                  >
                    sidebar.applicationInventory
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-1"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/reportsundefined"
                    tabindex="-1"
                  >
                    sidebar.reports
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-5"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/controls"
                    tabindex="-1"
                  >
                    sidebar.controls
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        <main
          class="pf-c-page__main"
          id="main-content-page-layout-horizontal-nav"
          tabindex="-1"
        />
      </div>
    </div>
    <div />
    <div />
    <div>
      <ul
        aria-live="polite"
        class="pf-c-alert-group pf-m-toast"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-c-page pf-m-resize-observer pf-m-breakpoint-default pf-m-height-breakpoint-sm"
    >
      <a
        class="pf-c-button pf-m-primary pf-c-skip-to-content"
        href="#main-content-page-layout-horizontal-nav"
      >
        Skip to content
      </a>
      <header
        class="pf-c-page__header"
      >
        <div
          class="pf-c-page__header-brand"
        >
          <div
            class="pf-c-page__header-brand-toggle"
          >
            <button
              aria-disabled="false"
              aria-expanded="false"
              aria-label="Global navigation"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="nav-toggle"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </button>
          </div>
          <a
            class="pf-c-page__header-brand-link"
          >
            <img
              alt="brand"
              class="pf-c-brand"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="pf-c-page__header-tools"
        >
          <div
            class="pf-c-page__header-tools-group pf-m-hidden"
          >
            <div
              class="pf-c-page__header-tools-item"
            >
              <button
                aria-disabled="false"
                aria-label="about-button"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                id="aboutButton"
                type="button"
              >
                <test-file-stub />
              </button>
            </div>
          </div>
          <div
            class="pf-c-page__header-tools-group"
          >
            <div
              class="pf-c-page__header-tools-item"
            >
              <div
                class="pf-c-dropdown pf-m-align-right"
                data-ouia-component-id="OUIA-Generated-Dropdown-1"
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Actions"
                  class="pf-c-dropdown__toggle pf-m-plain"
                  id="pf-dropdown-toggle-id-0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 192 512"
                    width="1em"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="pf-c-page__header-tools-item pf-m-hidden"
            />
          </div>
        </div>
      </header>
      <div
        aria-hidden="true"
        class="pf-c-page__sidebar pf-m-collapsed"
        id="page-sidebar"
      >
        <div
          class="pf-c-page__sidebar-body"
        >
          <nav
            aria-label="Nav"
            class="pf-c-nav"
            data-ouia-component-id="OUIA-Generated-Nav-1"
            data-ouia-component-type="PF4/Nav"
            data-ouia-safe="true"
            id="nav-primary"
          >
            <div
              class="perspective"
            >
              <div
                class="pf-c-select"
                data-ouia-component-id="OUIA-Generated-Select-single-1"
                data-ouia-component-type="PF4/Select"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-label="Options menu"
                  aria-labelledby=" pf-select-toggle-id-1"
                  class="pf-c-select__toggle"
                  id="pf-select-toggle-id-1"
                  type="button"
                >
                  <div
                    class="pf-c-select__toggle-wrapper"
                  >
                    <span
                      class="pf-c-select__toggle-text"
                    >
                      sidebar.developer
                    </span>
                  </div>
                  <span
                    class="pf-c-select__toggle-arrow"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <ul
              class="pf-c-nav__list"
              title="Global"
            >
              <li
                class="pf-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-1"
                data-ouia-component-type="PF4/NavItem"
                data-ouia-safe="true"
              >
                <a
                  class="pf-c-nav__link"
                  href="/applicationsundefined"
                  tabindex="-1"
                >
                  sidebar.applicationInventory
                </a>
              </li>
              <li
                class="pf-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-1"
                data-ouia-component-type="PF4/NavItem"
                data-ouia-safe="true"
              >
                <a
                  class="pf-c-nav__link"
                  href="/reportsundefined"
                  tabindex="-1"
                >
                  sidebar.reports
                </a>
              </li>
              <li
                class="pf-c-nav__item"
                data-ouia-component-id="OUIA-Generated-NavItem-5"
                data-ouia-component-type="PF4/NavItem"
                data-ouia-safe="true"
              >
                <a
                  class="pf-c-nav__link"
                  href="/controls"
                  tabindex="-1"
                >
                  sidebar.controls
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <main
        class="pf-c-page__main"
        id="main-content-page-layout-horizontal-nav"
        tabindex="-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
